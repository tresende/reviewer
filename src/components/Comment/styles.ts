import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  display: flex;
`

export const Avatar = styled.img`
  border-radius: 50%;
  height: 48px;
  width: 48px;
`

export const Comment = styled.div`
  ${({ theme }) => css`
    width: 100%;
    margin: 0 ${theme.spacings.xsmall};
    border-bottom: ${theme.border.default};
    border-color: ${theme.colors.gray100};
    margin-bottom: ${theme.spacings.small};
    padding-bottom: ${theme.spacings.small};
  `}
`

export const CommentHeader = styled.div`
  display: flex;
  justify-content: space-between;
  ${media.lessThan('medium')`
    flex-direction: column;
  `}
`

export const Name = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.bold};
    margin-right: ${theme.spacings.xxsmall};
    color: ${theme.colors.gray500};
  `}
`

export const SmallText = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.gray600};
    ${media.lessThan('medium')`
      display:block;
    `}
  `}
`

export const Text = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    margin-top: ${theme.spacings.xxsmall};
    min-height: 50px;
  `}
`
