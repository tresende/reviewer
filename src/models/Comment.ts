export default interface Comment {
  user: string
  photo: string
  text: string
  score: number
  date: Date
  id: number
}
