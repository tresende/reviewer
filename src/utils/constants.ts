const SCORE_LABEL = ['PÃ©ssimo', 'Ruim', 'Mediano', 'Bom', 'Perfeito']
const MIN_TEXT_LENGTH = 5
const MAX_RATING = 5
const DEFAULT_SCORE = 3
const ANIMATION_DURATION = 1000

export { SCORE_LABEL, MAX_RATING, ANIMATION_DURATION, MIN_TEXT_LENGTH, DEFAULT_SCORE }
