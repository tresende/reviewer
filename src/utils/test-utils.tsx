import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
