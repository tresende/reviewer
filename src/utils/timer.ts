export function timeout(ms?: number | undefined) {
  return new Promise((resolve) => setTimeout(resolve, ms))
}
