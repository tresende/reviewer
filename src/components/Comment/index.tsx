import Stars from 'components/Stars'
import Comment from 'models/Comment'
import { toPTBRFormat } from 'utils/date'

import * as S from './styles'

export type ButtonProps = {
  comment: Comment
}

const CommentComponent = ({ comment }: ButtonProps) => (
  <S.Wrapper>
    <S.Avatar src={comment.photo} />
    <S.Comment>
      <S.CommentHeader>
        <div>
          <S.Name>{comment.user}</S.Name>
          <S.SmallText>Avaliação verificada • </S.SmallText>
          <S.SmallText>{toPTBRFormat(comment.date)}</S.SmallText>
        </div>
        <Stars size={14} quantity={comment.score} />
      </S.CommentHeader>
      <S.Text>{comment.text}</S.Text>
    </S.Comment>
  </S.Wrapper>
)

export default CommentComponent
