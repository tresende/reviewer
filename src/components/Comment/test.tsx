import { render, screen } from 'utils/test-utils'
import Comment from 'models/Comment'
import Component from '.'

describe('<Comment />', () => {
  it('should render component correctly with mocked parameters', () => {
    const mock = {
      date: new Date(2019, 1, 2, 2, 2),
      photo: '',
      user: 'Thiago',
      text: 'SAMPLE',
      id: 1,
      score: 3
    } as Comment

    const { container } = render(<Component comment={mock} />)

    expect(screen.getByText('Thiago')).toBeInTheDocument()
    expect(screen.getByText('SAMPLE')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
