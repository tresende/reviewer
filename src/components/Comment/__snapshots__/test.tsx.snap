// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> should render component correctly with mocked parameters 1`] = `
.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 2.4rem;
}

.c7 {
  background-color: transparent;
  border: none;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  border-radius: 50%;
  height: 48px;
  width: 48px;
}

.c2 {
  width: 100%;
  margin: 0 1.6rem;
  border-bottom: 1px solid;
  border-color: #EEEEEE;
  margin-bottom: 2.4rem;
  padding-bottom: 2.4rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-size: 1.4rem;
  font-weight: 700;
  margin-right: 0.8rem;
  color: #4b4b4b;
}

.c5 {
  font-size: 1.2rem;
  color: #4A4A4A;
}

.c8 {
  font-size: 1.2rem;
  margin-top: 0.8rem;
  min-height: 50px;
}

@media (max-width:768px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c5 {
    display: block;
  }
}

<div
  class="c0"
>
  <img
    class="c1"
    src=""
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <div>
        <span
          class="c4"
        >
          Thiago
        </span>
        <span
          class="c5"
        >
          Avaliação verificada • 
        </span>
        <span
          class="c5"
        >
          02/02/2019
        </span>
      </div>
      <div
        class="c6"
        title="Mediano"
      >
        <button
          class="c7"
        >
          <svg
            color="#FAAF40"
            fill="currentColor"
            height="14"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(250, 175, 64);"
            viewBox="0 0 576 512"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
        </button>
        <button
          class="c7"
        >
          <svg
            color="#FAAF40"
            fill="currentColor"
            height="14"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(250, 175, 64);"
            viewBox="0 0 576 512"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
        </button>
        <button
          class="c7"
        >
          <svg
            color="#FAAF40"
            fill="currentColor"
            height="14"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(250, 175, 64);"
            viewBox="0 0 576 512"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
        </button>
        <button
          class="c7"
        >
          <svg
            color="#FAAF40"
            fill="currentColor"
            height="14"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(250, 175, 64);"
            viewBox="0 0 576 512"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
            />
          </svg>
        </button>
        <button
          class="c7"
        >
          <svg
            color="#FAAF40"
            fill="currentColor"
            height="14"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(250, 175, 64);"
            viewBox="0 0 576 512"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
            />
          </svg>
        </button>
      </div>
    </div>
    <p
      class="c8"
    >
      SAMPLE
    </p>
  </div>
</div>
`;
